micro-rdk/src/common/grpc.rs

```diff
@@ -1212,6 +1213,16 @@ where
         self.encode_message(operation)
     }
 
+    // robot_shutdown will not return anything because will restart
+    fn robot_shutdown(&mut self, _: &[u8]) -> ! {
+        #[cfg(feature = "native")]
+        std::process::exit(0);
+        #[cfg(feature = "esp32")]
+        unsafe {
+            crate::esp32::esp_idf_svc::sys::esp_restart();
```

Comment:
Is it OK that `shutdown` for an esp32 is just going to reboot and restart the robot? What are the intended semantics of `shutdown`? Should this require some more active measures to get the robot running again, like powercycling the esp32? We could, for instance, enter deep sleep with no wake source instead.